lib/entity.js

```diff
@@ -71,6 +86,21 @@ Entity.prototype.merge = function merge (snapshot) {
   return this;
 };
 
+/**
+ * Merge a property onto the instance.
+ *
+ * Given a name and a value, mergeProperty checks if the name corresponds to a
+ * function or an object.
+ * - If not either, the value is simply applied to the
+ * instance.
+ * - If value is an object and the name corresponds to a value that is
+ * also an object in the instance, it is merged.
+ * [CONFIRM] - If the name corresponds to a function, the funcition is executed with
+ * value passed as the parameter.
+ *
+ * @param  {String} name   the name of the property being merged.
+ * @param  {Object} value  the value of the property being merged.
+ */
 Entity.prototype.mergeProperty = function mergeProperty (name, value) {
   if (mergeProperties.size &&
```

Comment:
I'm not entirely sure how the mergeProperties Map is being used here. What the logic is behind this. Could you give a short explanation so I can add it to the docs?